google/cloud/internal/oauth2_service_account_credentials.cc

```diff
@@ -225,11 +225,13 @@ ServiceAccountCredentials::AuthorizationHeader() {
 }
 
 StatusOr<std::vector<std::uint8_t>> ServiceAccountCredentials::SignBlob(
-    std::string const& signing_account, std::string const& blob) const {
-  if (signing_account != info_.client_email) {
-    return Status(
-        StatusCode::kInvalidArgument,
-        "The current_credentials cannot sign blobs for " + signing_account);
+    absl::optional<std::string> const& signing_account,
+    std::string const& blob) const {
+  if (signing_account.has_value() &&
+      signing_account.value() != info_.client_email) {
```

Comment:
Nitty nit: Typically we use dereferencing instead of `value()` when it is a consequent of `has_value()`.